File: infra-market-quarkus/src/main/kotlin/io/infra/qk/controller/AuthController.kt
package io.infra.qk.controller

import io.infra.qk.dto.LoginRequest
import io.infra.qk.dto.ChangePasswordRequest
import io.infra.qk.dto.ApiResponse
import io.infra.qk.service.AuthService
import jakarta.inject.Inject
import jakarta.ws.rs.*
import jakarta.ws.rs.core.MediaType
import org.eclipse.microprofile.jwt.JsonWebToken

/**
 * 认证控制器
 * @author liuqinglin
 * Date: 2025/8/30
 */
@Path("/auth")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class AuthController @Inject constructor(
    private val authService: AuthService,
    private val jwt: JsonWebToken
) {
    
    @POST
    @Path("/login")
    fun login(request: LoginRequest): ApiResponse<*> {
        return authService.login(request)
    }
    
    @GET
    @Path("/current/user")
    fun getCurrentUser(): ApiResponse<*> {
        val userId = getCurrentUserId()
        return authService.getCurrentUser(userId)
    }
    
    @GET
    @Path("/user/menus")
    fun getUserMenus(): ApiResponse<*> {
        val userId = getCurrentUserId()
        return authService.getUserMenus(userId)
    }
    
    @POST
    @Path("/refresh/token")
    fun refreshToken(): ApiResponse<*> {
        val userId = getCurrentUserId()
        return authService.refreshToken(userId)
    }
    
    @POST
    @Path("/logout")
    fun logout(): ApiResponse<*> {
        val userId = getCurrentUserId()
        return authService.logout(userId)
    }
    
    @POST
    @Path("/change/password")
    fun changePassword(request: ChangePasswordRequest): ApiResponse<*> {
        val userId = getCurrentUserId()
        return authService.changePassword(userId, request)
    }
    
    private fun getCurrentUserId(): Long {
        return jwt.getClaim("userId") ?: 0L
    }
}

File: infra-market-server-dotnet/Controllers/AuthController.cs
using InfraMarket.Server.DTOs;
using InfraMarket.Server.Services.Interfaces;
using InfraMarket.Server.Utils;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace InfraMarket.Server.Controllers;

[ApiController]
[Route("auth")]
public class AuthController(
    IAuthService authService) : ControllerBase
{
    [HttpPost("login")]
    [AllowAnonymous]
    public async Task<ActionResult<ApiData<LoginResponse>>> Login([FromBody] LoginRequest request)
    {
        var result = await authService.LoginAsync(request);
        return Ok(result);
    }

    [HttpGet("current/user")]
    [Authorize]
    public async Task<ActionResult<ApiData<LoginResponse>>> GetCurrentUser()
    {
        var uid = User.GetUid();
        if (!uid.HasValue)
        {
            return Unauthorized(ApiData<LoginResponse>.Error("未登录", 401));
        }

        var result = await authService.GetCurrentUserAsync(uid.Value);
        return Ok(result);
    }

    [HttpGet("user/menus")]
    [Authorize]
    public async Task<ActionResult<ApiData<List<PermissionDto>>>> GetUserMenus()
    {
        var uid = User.GetUid();
        if (!uid.HasValue)
        {
            return Unauthorized(ApiData<List<PermissionDto>>.Error("未登录", 401));
        }

        var result = await authService.GetUserMenusAsync(uid.Value);
        return Ok(result);
    }

    [HttpPost("refresh/token")]
    [Authorize]
    public async Task<ActionResult<ApiData<LoginResponse>>> RefreshToken()
    {
        var uid = User.GetUid();
        if (!uid.HasValue)
        {
            return Unauthorized(ApiData<LoginResponse>.Error("未登录", 401));
        }

        var result = await authService.RefreshTokenAsync(uid.Value);
        return Ok(result);
    }

    [HttpPost("logout")]
    [Authorize]
    public async Task<ActionResult<ApiData<object>>> Logout()
    {
        var uid = User.GetUid();
        if (!uid.HasValue)
        {
            return Unauthorized(ApiData<object>.Error("未登录", 401));
        }

        var result = await authService.LogoutAsync(uid.Value);
        return Ok(result);
    }

    [HttpPost("change/password")]
    [Authorize]
    public async Task<ActionResult<ApiData<object>>> ChangePassword([FromBody] ChangePasswordRequest request)
    {
        var uid = User.GetUid();
        if (!uid.HasValue)
        {
            return Unauthorized(ApiData<object>.Error("未登录", 401));
        }

        var result = await authService.ChangePasswordAsync(uid.Value, request);
        return Ok(result);
    }

}
